lib/resty/luasocket/init.lua

```diff
@@ -139,6 +139,10 @@ do
         return return_bool and false or nil, err
       end
 
+      if verify then
+        sock:sni(server_name)
+      end
```

Comment:
The check should not be on `verify` but on whether an SNI was provided;
```suggestion
      if server_name then
        sock:sni(server_name)
      end
```